airflow/contrib/hooks/gcs_hook.py

```diff
@@ -356,38 +285,32 @@ def list(self, bucket, versions=None, maxResults=None, prefix=None, delimiter=No
         :type delimiter: str
         :return: a stream of object names matching the filtering criteria
         """
-        service = self.get_conn()
+        client = self.get_conn()
+        bucket = client.get_bucket(bucket)
 
-        ids = list()
+        ids = []
         pageToken = None
         while True:
-            response = service.objects().list(
-                bucket=bucket,
-                versions=versions,
-                maxResults=maxResults,
-                pageToken=pageToken,
+            blobs = bucket.list_blobs(
+                max_results=maxResults,
+                page_token=pageToken,
```

Comment:
This parameter is deprecated. Could you use a new way?

> page_token (str) – (Optional) If present, return the next batch of blobs, using the value, which must correspond to the `nextPageToken` value returned in the previous response. **Deprecated: use the pages property of the returned iterator instead of manually passing the token.**

Reply:
yes, that is in my todo list. As I wrote in the description of this PR, I want to keep this PR as backwards-compatible as possible, hence there is no note in Updating.md. I will add the note however as I think even though the function input and output are same, it still adds an extra dependency of `google-cloud-storage`, so I will do that.

I will take care of the Deprecated `nextPageToken` in the upcoming PR.